db/insertData.js

```diff
@@ -0,0 +1,74 @@
+const { Pool } = require('pg')
+const config = require('./config.json');
+const fs = require('fs');
+
+const pool = new Pool(config)
+
+let getData = (file, callback) => {
+  let dataInsert;
+  let start = Date.now()
+  fs.readFile(file, (err, data) => {
+    if (err) {
+      console.log('There was an error reading the file')
+    } else {
+      dataInsert = JSON.parse(data)
+      console.log('File was successfully read')
+      insertData(dataInsert, (number) => {
+        console.log(`Current entry: ${number}`)
+      }, callback, start)
+    }
+  })
+}
+
+let insertFunction = async (query, callback, time, i) => {
+  const client = await pool.connect()
+  await client.query(query)
+  if (i % 100000 === 0) {
```

Comment:
Remember that if statements are fairly expensive, especially when dealing with 1 - 10 million records. If you are batching your insertions, it may be worthwhile to get rid of these if statements and just log after every set of records is done inserting.

To clarify, my records are split into files containing 100,000 each, and I create a loop that opens a file, inserts everything, then moves onto the next file. Logging after every batch of records is cheaper than executing 2 if statements for every insertion. Just a thought that may help increase your insertion speed.